designs/SearchCommandSpec.md

```diff
@@ -0,0 +1,98 @@
+# Search CLI Tool for NuGet.exe and Dotnet.exe
+
+* Status: **Implementing**
+* Author(s): [Advay Tandon](https://github.com/advay26)
+* Issue: [9704](https://github.com/NuGet/Home/issues/9704) - Search CLI Tool for NuGet.exe and Dotnet.exe
+
+## Problem Background
+
+Search functionality from the command line for NuGet.exe and the .NET Core CLI has been a feature request from the NuGet open source community for some time. Additionally, the NuGet API is already equipped to handle search requests, so an extension of this service to command line users would be both practical to implement and also highly valuable.
+
+## Who are the customers
+
+All NuGet.exe and .NET CLI users.
+
+## Goals
+
+* Providing search functionality for NuGet packages through the commandline
+* Displaying the search results to users in a way that maximizes the visibility of the most relevant results.
+
+## Solution
+
+Th search feature leverages the search service provided by the NuGet API. Search targets multiple sources and displays all of their results one after the other. The command line interface is simple: __./NuGet.exe search \<query\>__.
+```
+PS C:\> ./NuGet.exe search logging
+```
+
+Each search result displays the name, version, number of downloads, and description of the package. Additionally, Search uses pagination to provide a clean viewing experience for users as they look at the results of their search. This ensures that users have access to the search results in order of decreasing relevance. A single page of results is shown at a time, and the user can look at the next page by pressing the __space__ key.
+
+```
+====================
+Source: NuGet.org
+--------------------
+> Microsoft.Extensions.Logging.Abstractions | v5.0.0-preview.6.20305.6 | DLs: 345145935
+Logging abstractions for Microsoft.Extensions.Logging.
+
+Commonly Used Types:
+Microsoft.Extensions.Logging.ILogger
+Microsoft.Extensions.Logging.ILoggerFactory
+Microsoft.Extensions.Logging.ILogger&lt;TCategoryName&gt;
+Microsoft.Extensions.Logging.LogLevel
+Microsoft.Extensions.Logging.Logger&lt;T&gt;
+Microsoft.Extensions.Logging.LoggerMessage
+Microsoft.Extensions.Logging.Abstractions.NullLogger
+
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging | v5.0.0-preview.6.20305.6 | DLs: 243186566
+Logging infrastructure default implementation for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.IdentityModel.Logging | v6.6.0 | DLs: 120177247
+Includes Event Source based logging support.
+
+--------------------
+> Microsoft.Extensions.Logging.Configuration | v5.0.0-preview.6.20305.6 | DLs: 63205816
+Configuration support for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging.Console | v5.0.0-preview.6.20305.6 | DLs: 61796431
```

Comment:
Will this show normalized version or full version (includes build metadata)? API returns full version.

Reply:
Looking at some specific examples, like NuGet.Protocol, the API seems to return the normalized version already, but feel free to correct me if I'm wrong.

Reply:
The NuGet.Protocol API returns whatever the HTTP endpoint returns. Per protocol specification, this should be the full version:
https://docs.microsoft.com/en-us/nuget/api/search-query-service-resource#search-result

I don't know the specific API you are using in client, but if a `NuGetVersion` is involved instead of a raw string, you can choose via one of these two methods:
https://github.com/NuGet/NuGet.Client/blob/088004379cb99890f766e9b7a0a29f7b080f888b/src/NuGet.Core/NuGet.Versioning/SemanticVersionBase.cs#L13-L30

Reply:
Ok, I'll make to sure to incorporate this into how we display results.